explainers/navigation-to-unsigned-bundles.md

```diff
@@ -84,6 +84,12 @@ that claims it.
 > An origin could use an "untrusted" bundle with same-origin resources to create
 > a kind of [suborigin](https://w3c.github.io/webappsec-suborigins/).
 
+The diagram below shows how signatures, flags, and primary URL of a WebBundle determine the mode in which the bundle is loaded.
+
+<img src="navigation-to-bundles.svg">
```

Comment:
We should be sure to commit the source for any graphics so they're easy to reproduce. e.g.

```suggestion
![graph TD
  s([Navigation to an<br/>application/webbundle<br/>resource]) -->
  p[Parse primary URL<br/>and control flags] -->
  Signed{Signed?}
  Signed -->|yes| outofscope(["Load as a signed bundle<br/>(Out of the scope<br/>of this document)"])
  Signed -->|no| Trusted{The trusted<br/>flag is set?}
  Trusted -->|yes| InScope{The primary URL is<br/>within the unsigned<br/>bundle scope?}
  Trusted -->|no| untrusted([Load as an<br/>untrusted<br/>bundle])
  InScope -->|yes| trust([Load as a<br/>trusted bundle])
  InScope -->|no| redirect([Redirect to the<br/>primary URL])
](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgcyhbTmF2aWdhdGlvbiB0byBhbjxici8-YXBwbGljYXRpb24vd2ViYnVuZGxlPGJyLz5yZXNvdXJjZV0pIC0tPlxuICBwW1BhcnNlIHByaW1hcnkgVVJMPGJyLz5hbmQgY29udHJvbCBmbGFnc10gLS0-XG4gIFNpZ25lZHtTaWduZWQ_fVxuICBTaWduZWQgLS0-fHllc3wgb3V0b2ZzY29wZShbXCJMb2FkIGFzIGEgc2lnbmVkIGJ1bmRsZTxici8-KE91dCBvZiB0aGUgc2NvcGU8YnIvPm9mIHRoaXMgZG9jdW1lbnQpXCJdKVxuICBTaWduZWQgLS0-fG5vfCBUcnVzdGVke1RoZSB0cnVzdGVkPGJyLz5mbGFnIGlzIHNldD99XG4gIFRydXN0ZWQgLS0-fHllc3wgSW5TY29wZXtUaGUgcHJpbWFyeSBVUkwgaXM8YnIvPndpdGhpbiB0aGUgdW5zaWduZWQ8YnIvPmJ1bmRsZSBzY29wZT99XG4gIFRydXN0ZWQgLS0-fG5vfCB1bnRydXN0ZWQoW0xvYWQgYXMgYW48YnIvPnVudHJ1c3RlZDxici8-YnVuZGxlXSlcbiAgSW5TY29wZSAtLT58eWVzfCB0cnVzdChbTG9hZCBhcyBhPGJyLz50cnVzdGVkIGJ1bmRsZV0pXG4gIEluU2NvcGUgLS0-fG5vfCByZWRpcmVjdChbUmVkaXJlY3QgdG8gdGhlPGJyLz5wcmltYXJ5IFVSTF0pXG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)
```

![graph TD
  s([Navigation to an<br/>application/webbundle<br/>resource]) -->
  p[Parse primary URL<br/>and control flags] -->
  Signed{Signed?}
  Signed -->|yes| outofscope(["Load as a signed bundle<br/>(Out of the scope<br/>of this document)"])
  Signed -->|no| Trusted{The trusted<br/>flag is set?}
  Trusted -->|yes| InScope{The primary URL is<br/>within the unsigned<br/>bundle scope?}
  Trusted -->|no| untrusted([Load as an<br/>untrusted<br/>bundle])
  InScope -->|yes| trust([Load as a<br/>trusted bundle])
  InScope -->|no| redirect([Redirect to the<br/>primary URL])
](https://mermaid.ink/img/eyJjb2RlIjoiZ3JhcGggVERcbiAgcyhbTmF2aWdhdGlvbiB0byBhbjxici8-YXBwbGljYXRpb24vd2ViYnVuZGxlPGJyLz5yZXNvdXJjZV0pIC0tPlxuICBwW1BhcnNlIHByaW1hcnkgVVJMPGJyLz5hbmQgY29udHJvbCBmbGFnc10gLS0-XG4gIFNpZ25lZHtTaWduZWQ_fVxuICBTaWduZWQgLS0-fHllc3wgb3V0b2ZzY29wZShbXCJMb2FkIGFzIGEgc2lnbmVkIGJ1bmRsZTxici8-KE91dCBvZiB0aGUgc2NvcGU8YnIvPm9mIHRoaXMgZG9jdW1lbnQpXCJdKVxuICBTaWduZWQgLS0-fG5vfCBUcnVzdGVke1RoZSB0cnVzdGVkPGJyLz5mbGFnIGlzIHNldD99XG4gIFRydXN0ZWQgLS0-fHllc3wgSW5TY29wZXtUaGUgcHJpbWFyeSBVUkwgaXM8YnIvPndpdGhpbiB0aGUgdW5zaWduZWQ8YnIvPmJ1bmRsZSBzY29wZT99XG4gIFRydXN0ZWQgLS0-fG5vfCB1bnRydXN0ZWQoW0xvYWQgYXMgYW48YnIvPnVudHJ1c3RlZDxici8-YnVuZGxlXSlcbiAgSW5TY29wZSAtLT58eWVzfCB0cnVzdChbTG9hZCBhcyBhPGJyLz50cnVzdGVkIGJ1bmRsZV0pXG4gIEluU2NvcGUgLS0-fG5vfCByZWRpcmVjdChbUmVkaXJlY3QgdG8gdGhlPGJyLz5wcmltYXJ5IFVSTF0pXG4iLCJtZXJtYWlkIjp7InRoZW1lIjoiZGVmYXVsdCJ9LCJ1cGRhdGVFZGl0b3IiOmZhbHNlfQ)


The PlantUML output is 1/10 the size...

```suggestion
![@startuml
:Navigation to an application/webbundle resource;
:Parse primary URL and control flags;
if (Signed?) then (yes)
  :Load as a signed bundle
  (Out of the scope of
  this document);
  stop
else (no)
  if (The trusted flag is set?) then (yes)
    if (The primary URL is\nwithin the unsigned\nbundle scope?) then (yes)
      :Load as a trusted bundle;
      stop
    else (no)
      :Redirect to the primary URL;
      stop
    endif
  else (no)
    :Load as an
    untrusted bundle;
    stop
  endif
endif
@enduml](http://www.plantuml.com/plantuml/svg/POyzRWCX48LxJl7ATPNUMyG7i9B8IJet66OT97POCKCitpuOP2KRDy2Wz_FDJjHcBNCqPljYlyFPQaWCJR0CkomnkFRpTA7JgR2FX4oIIdOqcksRpK9OSfXjlkBpiAyk3vTOSugOeZtBQCA4uJsScVpp1lf5ZE5AiZ70Tf-iXnLOI1EWLnXWU2sADDtq49SMgeD17OF09rTcOjsC1X1DYw4eX87JBVHMzr5Tceie-KQ1wXBI__rtyNg584U-XDh4hRrmPpjoX-iuZr6hTUxbtJ8sGMTjpp-ytNaW7p8vXIReckVHp3vCPXMoAkSs5tvW-0tf4VqqktgNEVu0)
```

![@startuml
:Navigation to an application/webbundle resource;
:Parse primary URL and control flags;
if (Signed?) then (yes)
  :Load as a signed bundle
  (Out of the scope of
  this document);
  stop
else (no)
  if (The trusted flag is set?) then (yes)
    if (The primary URL is\nwithin the unsigned\nbundle scope?) then (yes)
      :Load as a trusted bundle;
      stop
    else (no)
      :Redirect to the primary URL;
      stop
    endif
  else (no)
    :Load as an
    untrusted bundle;
    stop
  endif
endif
@enduml](http://www.plantuml.com/plantuml/svg/POyzRWCX48LxJl7ATPNUMyG7i9B8IJet66OT97POCKCitpuOP2KRDy2Wz_FDJjHcBNCqPljYlyFPQaWCJR0CkomnkFRpTA7JgR2FX4oIIdOqcksRpK9OSfXjlkBpiAyk3vTOSugOeZtBQCA4uJsScVpp1lf5ZE5AiZ70Tf-iXnLOI1EWLnXWU2sADDtq49SMgeD17OF09rTcOjsC1X1DYw4eX87JBVHMzr5Tceie-KQ1wXBI__rtyNg584U-XDh4hRrmPpjoX-iuZr6hTUxbtJ8sGMTjpp-ytNaW7p8vXIReckVHp3vCPXMoAkSs5tvW-0tf4VqqktgNEVu0)